components/tidb_query_datatype/src/macros.rs

```diff
@@ -0,0 +1,48 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+#[macro_export]
+macro_rules! other_err {
+    ($msg:tt) => ({
+        tidb_query_datatype::Error::from(tidb_query_datatype::error::EvaluateError::Other(
+            format!(concat!("[{}:{}]: ", $msg), file!(), line!())
+        ))
+    });
+    ($f:tt, $($arg:expr),+) => ({
+        tidb_query_datatype::Error::from(tidb_query_datatype::error::EvaluateError::Other(
+            format!(concat!("[{}:{}]: ", $f), file!(), line!(), $($arg),+)
+        ))
+    });
+}
+
+macro_rules! _other_err {
+    ($msg:tt) => ({
+        crate::Error::from(crate::error::EvaluateError::Other(
+            format!(concat!("[{}:{}]: ", $msg), file!(), line!())
+        ))
+    });
+    ($f:tt, $($arg:expr),+) => ({
+        crate::Error::from(crate::error::EvaluateError::Other(
+            format!(concat!("[{}:{}]: ", $f), file!(), line!(), $($arg),+)
+        ))
+    });
+}
```

Comment:
why two error?

Reply:
I should leave a comment here and marked the `_other_err` with `#[cfg(test)]`. Unit test can not  compile if use `tidb_query_datatype::Error`